ttljob: fix a range edge case

otan opened the pull request:
See individual commits for details.

Refs: #81208

otan committed:
ttljob: add debugging info

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/81207)
<!-- Reviewable:end -->

otan committed:
ttljob: fix a range edge case

Release note (bug fix): Fix a bug in row-level TTL where the last range
key of a table may overlap with a separate table or index, resulting in
a "error decoding X bytes" error message when performing row-level TTL.

rafiss reviewed (approved):
lgtm.. but there probably  are other folks out there who know range encoding better than me, so idk if there's a different way

otan commented:
i looked at how SHOW RANGES FOR TABLE worked and uh yeah this is what i found :D!

bors r=rafiss
